server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/EventDeadLettersRedeliverTask.java

```
@@ -0,0 +1,119 @@
+                              Optional<Group> group, Optional<Event.EventId> eventId) {
+            this.successfulRedeliveriesCount = successfulRedeliveriesCount;
+            this.failedRedeliveriesCount = failedRedeliveriesCount;
+            this.group = group;
+            this.eventId = eventId;
+        }
+
+        public long getSuccessfulRedeliveriesCount() {
+            return successfulRedeliveriesCount;
+        }
+
+        public long getFailedRedeliveriesCount() {
+            return failedRedeliveriesCount;
+        }
+
+        @JsonInclude(JsonInclude.Include.NON_ABSENT)
+        public Optional<String> getGroup() {
+            return group.map(Group::asString);
+        }
+
+        @JsonInclude(JsonInclude.Include.NON_ABSENT)
+        public Optional<String> getEventId() {
+            return eventId.map(eventId -> eventId.getId().toString());
+        }
+    }
+
+    private final EventDeadLettersRedeliverService service;
+    private final EventRetriever eventRetriever;
+    private final AtomicLong successfulRedeliveriesCount;
+    private final AtomicLong failedRedeliveriesCount;
+
+    EventDeadLettersRedeliverTask(EventDeadLettersRedeliverService service, EventRetriever eventRetriever) {
+        this.service = service;
+        this.eventRetriever = eventRetriever;
+        this.successfulRedeliveriesCount = new AtomicLong(0L);
+        this.failedRedeliveriesCount = new AtomicLong(0L);
+    }
+
+    @Override
+    public Result run() {
+        return service.redeliverEvents(eventRetriever)
+            .map(this::updateCounters)
+            .reduce(Result.COMPLETED, Task::combine)
+            .block();
+    }
+
+    private Result updateCounters(RedeliverResult redeliverResult) {
+        switch (redeliverResult) {
+            case REDELIVER_SUCCESS:
+                successfulRedeliveriesCount.incrementAndGet();
+                return Result.COMPLETED;
+            case REDELIVER_FAIL:
+            default:
```

IMO this is a bad way of handling default. Generaly failing early on switch mismatch is better. Can we throw an exception instead?